src/lib/components/TokenSelect/index.tsx

```diff
@@ -72,45 +88,59 @@ export function TokenSelectDialog({ value, onSelect }: TokenSelectDialogProps) {
 
   const [options, setOptions] = useState<ElementRef<typeof TokenOptions> | null>(null)
 
+  const listHasMoreTokensThanJustNativeAsset = useMemo(() => {
+    if (tokens.length > 1) {
+      return true
+    }
+    return !tokens.find((token) => token.isNative)
+  }, [tokens])
+
   return (
     <>
       <Header title={<Trans>Select a token</Trans>} />
-      <Column gap={0.75}>
-        <Row pad={0.75} grow>
-          <ThemedText.Body1>
-            <SearchInput
-              value={query}
-              onChange={setQuery}
-              placeholder={t`Search by token name or address`}
-              onKeyDown={options?.onKeyDown}
-              onBlur={options?.blur}
-              ref={input}
-            />
-          </ThemedText.Body1>
-        </Row>
-        {Boolean(baseTokens.length) && (
-          <Row pad={0.75} gap={0.25} justify="flex-start" flex>
-            {baseTokens.map((token) => (
-              <TokenBase value={token} onClick={onSelect} key={currencyId(token)} />
-            ))}
-          </Row>
-        )}
-        <Rule padded />
-      </Column>
-      {isLoaded ? (
-        tokens.length ? (
-          <TokenOptions tokens={tokens} onSelect={onSelect} ref={setOptions} />
-        ) : (
-          <Column padded>
-            <Row justify="center">
-              <ThemedText.Body1 color="secondary">
-                <Trans>No results found.</Trans>
+      {listHasMoreTokensThanJustNativeAsset || !isLoaded ? (
```

Comment:
This inline logic is getting hard to follow. Can you break it into a
```
const Body = useCallback(() => {
  if (listHasTokens) {
    ...
  }
  return <NoTokensAvailableOnNetwork />
}, [])
```

Reply:
Alternatively, you can move the Header into the parent component, and have this entire componet just be the body to simplify a little. Then you can have early returns. I think that'd make this component a lot easier to read.